plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveTableHandle.java

```diff
@@ -345,7 +359,20 @@ public String getTableName()
         return dataColumns;
     }
 
-    // do not serialize partitions as they are not needed on workers
+    /**
+     * Represents raw partition information as String
```

Comment:
Is it filtered by table predicate?
Can `partitionNames` be loaded independently of  `partitions`?

Reply:
> Is it filtered by table predicate?

Yes but partially. 

> Can partitionNames be loaded independently of partitions?

If partitions is loaded then partitionNames is reset to `Optional.empty`

Reply:
Could you add that as a comment?

Reply:
> > Is it filtered by table predicate?
> 
> Yes but partially.

that must be documented